graphql_compiler/global_utils.py

```diff
@@ -29,3 +31,16 @@ def is_same_type(left, right):
         return is_same_type(left.of_type, right.of_type)
     else:
         return False
+
+
+def generate_new_name(base_name: str, taken_names: Sequence[str]) -> str:
```

Comment:
This approach seems unnecessarily wasteful to me. Imagine that `len(taken_names) == N`. The next call to `generate_new_name` will take `O(N^2)` to generate a single new name:
- The `while` loop will only break out on iteration number `N + 1`.
- Since `taken_names` is a `Sequence`, the `in` check in the `while` loop is `O(N)`.

Why not use an infinite generator of new names, where each `next()` call on the generator produces a new, unique name? Or at the very least, why not make `taken_names` be a `Set`, and make the starting index be `len(taken_names)`?

Reply:
Yes, but that is a wasteful use of this function. The solution is not to use it. I don't see an easy general solution, and I'd rather take this function out of `global_utils.py` than spend more work on it.

Reply:
Sounds good, let's do that then.